core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -940,17 +959,17 @@ object IO extends Serializable {
   /**
    * Returns an `IO` that is interrupted.
    */
-  final def interrupt: IO[Nothing, Nothing] = fail0(Cause.interrupted)
+  final def interrupt: IO[Nothing, Nothing] = fail(Cause.interrupted)
 
   /**
    * Returns an `IO` that terminates with the specified `Throwable`.
    */
-  final def terminate(t: Throwable): IO[Nothing, Nothing] = fail0(Cause.unchecked(t))
+  final def raiseUnchecked(t: Throwable): IO[Nothing, Nothing] = fail(Cause.unchecked(t))
```

Comment:
👎 on this as well

Reply:
no opinion from my side -- just following the description of the ticket

Reply:
@kaishh `raiseChecked` / `raiseUnchecked` have symmetry and are similar to Cats `raiseError`. 

I do like `fail`, personally. I feel the main goal should be symmetry, e.g. 

```scala
IO.success / IO.failure
IO.succeed / IO.fail
```

What do you think? /cc @sideeffffect

Reply:
@kaishh 

The reason I suggested `fail` is because there is symmetry with `Exit`. Every fiber either `ExitResult.Success` or `ExitResult.Failure`. These correspond (or could / should correspond) to `IO.success` / `IO.failure`, or `IO.succeed` / `IO.fail`. In other words, the structure of the `Exit` should  mirror the `IO`.

Reply:
I like `raiseChecked` / `raiseUnchecked` because:
 * symmetry
 * nicely corresponds to cats' `raiseError`, but we have 2, one for "checked exceptions" and the other for "unchecked exceptions"
 * and most importantly, because we're dealing with behaviour analogous to throwing exception, either `throw` or `raise` keywords should be used (so `throwChecked` / `throwUnchecked` would also be cool, maybe even better, because Scala itself uses `throw`)

Reply:
I like symmetry though I might try to invoke it to argue for the other side - e.g. one of these is not like the others: `terminate`/`unchecked` puts you outside of the realm Bifunctor IO and into the realm of Java Throwable, multiple error states, etc. 
So, maybe `succeed`/`fail` could be symmetric with each other while leaving unchecked/terminate/raise/panic/die/etc. with a name that stands out?

Reply:
Agree with @kaishh , raiseUnchecked sound very Java-ish and exception-related even though we're not using exception as the error type (E can be anything).

When dealing with an IO[E, A] it sounds pretty symmetric to me to use `IO.succeed` to return an A and `IO.fail` to return an E. Those should be used most of the time and stay simple.

On the other hand it makes sense to have something more scary with an exception-related word to replace `IO.terminate` since you actually need to provide an exception for it.

Reply:
> raiseUnchecked sound very Java-ish and exception-related

I'm not sure I would agree with that Java part (Java itself uses `throw` like Scala, not `raise` like F# for example). On other hand it I agree it does sound *exception-related*, which is good I think, because

> exception as the error type (E can be anything)

even though you're right that the `E` part in `IO[E, A]` doesn't have to extend `java.lang.Exception` it's not completely arbitrary. `IO` is right-biased after-all and we treat `E` as a form of a checked exception. and it's in that light where `raiseChecked` (and it's unchecked counterpart `raiseUnchecked`) make the most sense. and I suspect that *new coming users will have easier time with this naming convention* right when they'll be exploring the methods in an IDE.

with `raiseChecked` / `raiseUnchecked` it's obvious just from the names what's going on. with `fail` / `terminate` there will have to be an explanation which is for checked exceptions and which is for unchecked. tbh I'm not the biggest fan of `IO.suceed` -- I think there should be only `now` (`A`) and `sync` (`=> A`). and no `point` -- it's misleading name for sideeffectful code and it's AFAIK `A` (not `=> A`) in ScalaZ 8 anyway, but that's for a different discussion.

Reply:
that all being said, I'm not that attached to the new names :D I just followed the description of the ticket, and definitely can change the PR, no problem
@jdegoes @regiskuckaertz @kaishh @ghostdogpr guys can you please decide this among yourselves?

Reply:
How about these:

```scala
IO.fail0 => IO.halt
IO.fail => IO.fail
IO.point => IO.succeed
IO.now => IO.succeedNow
IO.terminate => IO.die
```

I also think the `sync*` family has grown too complex and I'd like to simplify that, but it will require engineering to achieve something like `syncTry[Throwable](...)` so separate PR.

Reply:
@jdegoes please pardon my ignorance -- what is the current `IO.point` good for anyway? shouldn't the default way of lifting a pure value to `IO` be `IO.now`? and thus possible renaming scheme would be `IO.now => IO.succeed`?
maybe I'm missing something, so thanks for any explanation :)

Reply:
It's important that `point` be lazy, but it's certainly true that _if_ `now` is more common than `point`, we can give the shorter name to `now`.

Looking through ZIO, I see:

 - 290 occurrences of `IO.now`
 - 80 occurrences of `IO.point`

While not a large sample or anything, this does suggest `IO.now` could be renamed to `IO.succeed`, and `IO.point` renamed to `IO.succeedLazy` (?).

Reply:
I also use `now` way more than `point`. Very much like the `fail`/`die` proposal as well 👍